Sources/TranscribeCLI/main.swift

```diff
@@ -3,42 +3,56 @@
 ///
 /// Usage:
 ///
-///     TranscribeCLI FILE
+///     TranscribeCLI --json FILE
+///     TranscribeCLI --json FILE --names SPEAKER SPEAKER
 ///
 ///     FILE: The input file. Must be in Amazon Transcribe JSON format.
+///     SPEAKER: Name of the speaker. Should be ordered by appearance.
```

Comment:
I'm not a fan of this being an array "ordered by appearance". I think this should be a dictionary (i.e., key-value pairs). So people should pass in something like this:

```
... --speakers spk_0="Alice Adams" spk_1="Bob Beetle"
```

This gives people more control and makes the mapping clearer. Not sure how if SwiftPM's argument parser supports this format.

Reply:
Why do you want to stick to the "spk_X" pattern in the CLI? 🤔
I see the labels "Speaker X" in the screenshot of the [SwiftCommunityPodcast Issue](https://github.com/SwiftCommunityPodcast/podcast/issues/15). But that gives you the correct order, right?
I am just not sure what additional information can be found in this mapping.

Reply:
> Why do you want to stick to the "spk_X" pattern in the CLI?

To me, the `"spk_0"`, `"spk_1"` strings are like speaker IDs. (I called them `Speaker.label` in the code; maybe `Speaker.id` would be the better name.) It seems logical (to me) to use these IDs as the keys by which we identify speakers and assign them their real names.

> I see the labels "Speaker X" in the screenshot of the [SwiftCommunityPodcast Issue](https://github.com/SwiftCommunityPodcast/podcast/issues/15). But that gives you the correct order, right?

Actually, I think the screenshot is a good counterexample. If you listen to the podcast episode, John spoke first, then Garric, and then Chris. So I (naturally?) ran the CLI program like this:

```sh
swift run TranscribeCLI \ 
  --json TestFixtures/amazon-transcribe-swift-community-podcast-0001.json \ 
  --names "John Sundell" "Garric Nahapetian" "Chris Lattner" "(Unknown)"
```

When you do that, you'll get the wrong speaker mapping! The beginning of the file looks like this:

```
WEBVTT

00:00:00.540 --> 00:00:21.250
<v Garric Nahapetian>Welcome to the very first episode of the Swift Community Podcast. A show for the Swift community by the Swift Community. I am one of your hosts, John Sindel. And with me, I have two wonderful co hosts, the first of which you might know as the host of the Swift coders podcast. Mr. Garric Nahapetian. How's going Garric?

00:00:21.380 --> 00:00:24.270
<v Chris Lattner>Excellent. Thank you so much for having me. How you doing?
...
```

So the code shifted every speaker one position to the left (and John is now speaker 4). The correct order on the command line is:

```sh
  --names "(Unknown)" "John Sundell" "Garric Nahapetian" "Chris Lattner"
```

This is because the `"speaker_label"`s (or IDs) in the JSON file are actually "spk_0", "spk_1", "spk_2", and "". And the code sorts these so that "" (the empty string) becomes the first item in the array.

I think there are two issues here:

1. An empty string should not be a valid speaker ID/label. Amazon Transcribe assigns this to empty paragraphs. The code should detect this and delete those segments (and the corresponding `Speaker` value outright). Our sample episode should have 3 speakers, not 4. We should probably consider this a bug.

2. Fixing the bug would solve the immediate issue, but I'd still prefer the mapping between speaker ID and speaker name to be explicit. The IDs/labels are what's in the JSON source file. For example, you might want to process a partial source file (where the first paragraph is missing or so) and this shouldn't cause the speaker mappings to shift around because the previously second speaker in the array is now the one who speaks first.

Reply:
Ah ok, now I can see your point! Thank you for this explanation